source/slang/slang-emit-glsl.cpp

```diff
@@ -1470,6 +1470,19 @@ void GLSLSourceEmitter::emitEntryPointAttributesImpl(
     default:
         break;
     }
+
+    if (irFunc->findDecoration<IRQuadDerivativesDecoration>())
+    {
+        m_writer->emit("layout(quad_derivatives) in;\n");
+    }
+    if (irFunc->findDecoration<IRRequireFullQuadsDecoration>())
```

Comment:
This should be moved to fragment shader specific handling logic in the switch statement above since it is only valid on fragment shaders.